<RadioGroup /> - add `selectionAreaSkin` and `selectionAreaPadding` props

assafganor opened the pull request:
<!---
Thanks for submitting a pull request 😄 !
-->

<!---
- Be as descriptive as possible when explaining what was changed.
- Link to an issue if one exists
-->

### 🔦 Summary
New props for `selectionArea`:
`selectionAreaSkin` and `selectionAreaPadding` 

https://app.zeplin.io/project/5c8f6f1f9f489505cac756dc/screen/5c9b502feb413e0ab95466be
...

<!--- Please mark all checkbox. If one is not relevant - delete it -->

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers <!--- Please tag the relevant team member -->
- [x] 👨‍🎨 UX change is approved by the Design System UX <!--- Please tag the relevant team member -->
- 📚 Change is documented
  - [x] Story
  - [x] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

WixBuildServer commented:
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6408

EgoziE commented:
@assafganor Hey
Please pay attention to visual tests, it seems there are extra margin/padding somewhere
![image](https://user-images.githubusercontent.com/35173937/102468190-e39e4700-4059-11eb-9b24-137ad08265ca.png)

EgoziE commented on src/RadioGroup/RadioButton/RadioButton.js:
```diff
@@ -95,34 +102,42 @@ class RadioButton extends React.PureComponent {
-            className={classnames({
-              [classes.vcenter]: vAlign === 'center',
+            className={classnames(classes.label, {
+              [classes.disabled]: disabled,
+              [classes.checked]: checked,
```
These states exist in a higher element already. (classes.radioWrapper)

assafganor replied:
removed and updated css

EgoziE commented on src/RadioGroup/RadioButton/RadioButton.js:
```diff
@@ -95,34 +102,42 @@ class RadioButton extends React.PureComponent {
-              style={{ height: lineHeight }}
-              className={classes.radioButtonWrapper}
-              data-hook={dataHooks.RadioButtonRadio}
+              className={classnames(classes.labelInner, {
+                [classes.vcenter]: vAlign === 'center',
+              })}
+              style={{ padding: selectionAreaPadding }}
```
Is this extra div mandatory?
Padding can be part of the label itself.

assafganor replied:
if padding will be part of label , the padding that I added for the outlined selectionAreaSkin will be override.
I needed to separate the `selectionAreaPadding` from the `selectionAreaSkin` that adds border/padding to label

EgoziE replied:
I see what you mean.
Well, in order to remove the extra div and for the CSS to be readable and understandable without padding games, in my opinion the best solution here is to have a permanent border for all states with a relevant color for each state.

EgoziE replied:
What do you think?
(same goes for the Checkbox PR)

assafganor replied:
It's not only color, we still need padding/border games because
default is:
border 1px , padding: 1px
and checked is:
border: 2px , padding: 0

Maybe I'll use variables for the padding and border values?
$SelectionAreaBorderWidth: '1px'
$SelectionAreaPadding: '1px'
$SelectionAreaBorderWidthChecked: '2px'
$SelectionAreaPaddingChecked: '0px'

assafganor replied:
@EgoziE  Do you have other idea how to make it more readable? or how can we do it without border/padding games?
I thought about `box-sizing: border-box` but we don't know the width and height.

EgoziE replied:
I understand the complexity of this design.
In order not to block you, let's take this solution with the extra div, we'll probably refactor the component sometime soon to remove classnames library anyway.

milkyfruit reviewed (commented):
@assafganor `selectionAreaSkin="outlined"` looks great.
It's strange that `selectionAreaPadding` when set to `1px` it's not 1 but 15 pixels. Why it's implemented this way? I think we should give a full control.

assafganor commented:
@milkyfruit 
I thought about our needs and missed your instructions in Zeplin, so I just added padding to the default values, instead of override it completely.

milkyfruit commented:
> @milkyfruit
> I thought about our needs and missed your instructions in Zeplin, so I just added padding to the default values, instead of override it completely.

@assafganor I think it should let to override all the padding, otherwise prop is confusing.

assafganor commented:
> > @milkyfruit
> > I thought about our needs and missed your instructions in Zeplin, so I just added padding to the default values, instead of override it completely.
> 
> @assafganor I think it should let to override all the padding, otherwise prop is confusing.

I agree, I fixed it
